auth/utils/file-reading-helpers.ts

```diff
@@ -59,11 +62,33 @@ export function fileExistsAtPath(filepath): boolean {
   return false;
 }
 
-export function constructFilepath(filepath): string {
+export function constructFilepath(filepath: string): string {
   // ensure filepath includes the filename
-  if (!filepath.endsWith(filename)) {
-    filepath = path.join(filepath, filename);
+  if (!filepath.endsWith(defaultCredsFilename)) {
+    filepath = path.join(filepath, defaultCredsFilename);
   }
 
   return filepath;
 }
+
+export function readCrTokenFile(filepath: string): string {
+  if (!fs.existsSync) {
+    return '';
+  }
+
+  let token: string = '';
+  if (fileExistsAtPath(filepath)) {
+    token = fs.readFileSync(filepath, 'utf8');
+    logger.debug(`Successfully read CR token from file: ${filepath}`);
+  } else {
+    logger.error(`Expected to find CR token file but the file does not exist: ${filepath}`);
+    throw new Error(`File does not exist: ${filepath}`);
+  }
+
+  if (token === '') {
```

Comment:
In the other SDKs we don't check for empty file. I think we should be consistent across the languages, so I will happily add this to the Python core if you wish.

Reply:
That's fair - I saw @padamstx comment on your PR about having one error to capture the "can't read token" scenario. Perhaps I will update this code to only throw one generic error, but leave in the more specific log messages to help the user in the case of an issue.

What do you think about that?

Reply:
Sounds good to me!

Reply:
Okay, update made 👍